sdk/servicebus/service-bus/test/batchReceiver.spec.ts

```diff
@@ -1190,4 +1191,245 @@ describe("Batching - disconnects", function(): void {
     settledMessageCount.should.equal(2, "Unexpected number of settled messages.");
     refreshConnectionCalled.should.be.greaterThan(0, "refreshConnection was not called.");
   });
+
+  it("returns messages if drain is in progress (receiveAndDelete)", async function(): Promise<
+    void
+  > {
+    // Create the sender and receiver.
+    sbClient = getServiceBusClient();
+    const { receiverClient, senderClient } = await getSenderReceiverClients(
+      sbClient,
+      TestClientType.UnpartitionedQueue,
```

Comment:
What about the sessions?
(Not asking about the tests)

Reply:
Sessions doesnt have the previous set of improvements we did for batch operations mainly because on connection disconnect event, we dont bring back any of the session receivers. We don't close them either. They would need a deeper look and slightly separate treatment, so keeping it outside the scope of this PR.

Reply:
Logged https://github.com/Azure/azure-sdk-for-js/issues/8875 for follow up